python/openassetio/specification/__init__.py

```diff
@@ -0,0 +1,21 @@
+#
+#   Copyright 2013-2021 The Foundry Visionmongers Ltd
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""
+Entity type system and property transport mechanism.
+"""
+__all__ = ["BlobSpecification", "Specification"]
+
+from openassetio._openassetio.specification import BlobSpecification, Specification
```

Comment:
This came from an annoyance with my IDE, which is also now causing problems with `pylint` for a similar reason.  

The `python/openassetio` source directory doesn't contain `_openassetio.so` (since it's installed into the venv), so relative imports fail, tripping up my IDE code completion.  I attempted to work around this by making it an absolute import.  So I made it absolute just for selfish reasons and forgot about it.  

However, now with `pylint` the same problem occurs. Linting fails because of unrecognized imports, even if it's executed from the venv where `_openassetio.so` is installed.  This is apparently because the source vs. installed `openassetio` package names conflict (as evidenced by the fact that `pylint` on the `test` directory succeeds).

Placing `_openassetio.so` alongside, rather than within, the `openassetio` package would make dev life much easier on several fronts.  E.g. we may be able to re-introduce `--editable` installs for running tests, speeding up the TDD cycle.

I understand that we don't want to structure our package based on what's convenient for dev, though. 

We could run `pylint` on the installed package, working around that problem, but IDE code completion would still suffer.

We could "install" the Python extension module to the source directory, so that relative imports work in IDE/pylint, then add it as an installable artefact in `setup.py`, which may be the correct thing to do anyway (I don't know much about `setup.py` and extension modules).

Reply:
In the meantime we can switch to absolute imports and disable the pylint checks as a quick fix rather than worrying about it in this PR.  But I would really like to solve this problem sooner rather than later - having good code completion (and thorough linting) is extremely useful.

Reply:
Oh man, that sounds like a _load_ of fun. Lets scheme on this sooner rather than later.

Reply:
I disabled the pylint checks back in 4348ffb0 - though none of the Github checks are running at the moment for some reason :thinking: